parent/pom.xml

```diff
@@ -1313,8 +1314,6 @@
             <id>checkstyle</id>
 
             <activation>
-                <activeByDefault>true</activeByDefault>
-
                 <property>
                     <name>skipTests</name>
```

Comment:
Does this property definition means that when `-DskipTests` is called, `checkstyle` profile will be activated?
And if so - what's the point of this action? By calling `-DskipTests` we usually trying to skip as much of non build related steps as possible.

Reply:
Yes, you are right. It's a discussable thing, but from my point, the style checking should be related to the build procedure.

Reply:
Then I would insists to use it like `<name>!skipTests</name>`
In that case it will be activated every time when `-DskipTests` is not specified (because that flag used to quickly build project without tests, docs, and other stuff).

Reply:
Petr,
Actually, it's not always true. 
`-DskipTests` - will compile source code, javadoc, skip test execution
`-Dmaven.javadoc.skip=true` - additionally will skip javadoc
`-Dmaven.test.skip=true` -- will skip tests compiling

So, I'd like to keep it as it is (enabled by default) and if we need a quick build - skip the checkstyle plugin by disabling it expliciy. For instance,

`mvn clean package -DskipTests -P !checkstyle`

http://maven.apache.org/plugins-archives/maven-surefire-plugin-2.12.4/examples/skipping-test.html

Reply:
I cannot agree.
Checkstyle is another test test that count towards PR merge eligibility. With -DskipTests it should be skipped as well.

Reply:
It's not a test, it's a code validation step and run before source code compilation. `skipTests` is related to the JUnit tests execution only and nothing more. 

As I've previously mentioned it's a discussable thing. Let's discuss it on the appropriate topic on dev-list.